metadata-io/src/main/java/com/linkedin/metadata/timeline/eventgenerator/GlossaryTermInfoChangeEventGenerator.java

```diff
@@ -0,0 +1,113 @@
+        String targetDescription = (targetDatasetProperties != null) ? targetDatasetProperties.getDefinition() : null;
+
+        if (baseDescription == null && targetDescription != null) {
+            // Description added
+            changeEvents.add(ChangeEvent.builder().entityUrn(entityUrn)
+                    .category(ChangeCategory.DOCUMENTATION)
+                    .operation(ChangeOperation.ADD)
+                    .semVerChange(SemanticChangeType.MINOR)
+                    .description(String.format(DESCRIPTION_ADDED, entityUrn, targetDescription))
+                    .auditStamp(auditStamp)
+                    .build());
+        } else if (baseDescription != null && targetDescription == null) {
+            // Description removed.
+            changeEvents.add(ChangeEvent.builder()
+                    .entityUrn(entityUrn)
+                    .category(ChangeCategory.DOCUMENTATION)
+                    .operation(ChangeOperation.REMOVE)
+                    .semVerChange(SemanticChangeType.MINOR)
+                    .description(String.format(DESCRIPTION_REMOVED, entityUrn, baseDescription))
+                    .auditStamp(auditStamp)
+                    .build());
+        } else if (baseDescription != null && targetDescription != null && !baseDescription.equals(targetDescription)) {
+            // Description has been modified.
+            changeEvents.add(ChangeEvent.builder()
+                    .entityUrn(entityUrn)
+                    .category(ChangeCategory.DOCUMENTATION)
+                    .operation(ChangeOperation.MODIFY)
+                    .semVerChange(SemanticChangeType.MINOR)
+                    .description(String.format(DESCRIPTION_CHANGED, entityUrn, baseDescription, targetDescription))
+                    .auditStamp(auditStamp)
+                    .build());
+        }
+        return changeEvents;
+    }
+
+    @Nullable
+    private static GlossaryTermInfo getGlossaryTermInfoFromAspect(EntityAspect entityAspect) {
+        if (entityAspect != null && entityAspect.getMetadata() != null) {
+            return RecordUtils.toRecordTemplate(GlossaryTermInfo.class, entityAspect.getMetadata());
+        }
+        return null;
+    }
+
+    @Override
+    public ChangeTransaction getSemanticDiff(EntityAspect previousValue, EntityAspect currentValue,
+                                             ChangeCategory element, JsonPatch rawDiff, boolean rawDiffsRequested) {
+        if (!previousValue.getAspect().equals(GLOSSARY_TERM_INFO_ASPECT_NAME) || !currentValue.getAspect()
+                .equals(GLOSSARY_TERM_INFO_ASPECT_NAME)) {
+            throw new IllegalArgumentException("Aspect is not " + GLOSSARY_TERM_INFO_ASPECT_NAME);
+        }
+        List<ChangeEvent> changeEvents = new ArrayList<>();
+        if (element == ChangeCategory.DOCUMENTATION) {
+            GlossaryTermInfo baseGlossaryTermInfo= getGlossaryTermInfoFromAspect(previousValue);
```

Comment:
```suggestion
            GlossaryTermInfo baseGlossaryTermInfo = getGlossaryTermInfoFromAspect(previousValue);
```

Checkstyle failing, otherwise looks good!